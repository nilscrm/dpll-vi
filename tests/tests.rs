use insta_cmd::assert_cmd_snapshot;
use rstest::rstest;
use std::{fs::File, path::{Path, PathBuf}, process::Command};

fn test_cnf_files(group_name: &str, path: &Path) {
  insta::with_settings!({
    snapshot_path => format!("snapshots/{}", group_name),
    filters => [(r"Performance\s+Time\s+.*\sms\s+Speed.*IPS", "[Performance...]")] },
  {
    let stdin = File::open(path).unwrap();
    assert_cmd_snapshot!(path.file_stem().unwrap().to_str().unwrap(), Command::new("vine").args(["run", "dpll/main.vi"]).arg("--breadth-first").stdin(stdin));
  });
}

#[rstest]
fn test_minimal(#[files("tests/minimal/*.cnf")] path: PathBuf) {
  test_cnf_files("minimal", &path);
}

#[rstest]
fn test_medium(#[files("tests/medium/*.cnf")] path: PathBuf) {
  test_cnf_files("medium", &path);
}

#[rstest]
fn test_satlib_sat(#[files("tests/uf5-218_first_10/*.cnf")] path: PathBuf) {
  test_cnf_files("uf5-218_first_10", &path);
}

#[rstest]
fn test_satlib_unsat(#[files("tests/uuf50-218_first_10/*.cnf")] path: PathBuf) {
  test_cnf_files("uuf50-218_first_10", &path);
}
