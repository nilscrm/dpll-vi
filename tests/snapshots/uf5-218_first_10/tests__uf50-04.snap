---
source: tests/tests.rs
info:
  program: vine
  args:
    - run
    - dpll/main.vi
    - "--breadth-first"
---
success: true
exit_code: 0
----- stdout -----
satisfiable
evaluation under model: true
var 1 = true
var 2 = true
var 3 = false
var 4 = true
var 5 = true
var 6 = false
var 7 = false
var 8 = true
var 9 = true
var 10 = true
var 11 = false
var 12 = true
var 13 = true
var 14 = false
var 15 = false
var 16 = false
var 17 = true
var 18 = false
var 19 = false
var 20 = false
var 21 = true
var 22 = true
var 23 = true
var 24 = true
var 25 = true
var 26 = false
var 27 = true
var 28 = false
var 29 = false
var 30 = false
var 31 = true
var 32 = false
var 33 = false
var 34 = false
var 36 = false
var 37 = false
var 38 = true
var 40 = false
var 41 = true
var 42 = false
var 43 = false
var 44 = true
var 45 = false
var 46 = true
var 47 = false
var 48 = true
var 49 = false
var 50 = false

----- stderr -----

Interactions
  Total           63_309_131
    Depth         27_818_428
    Breadth                2
  Annihilate      35_934_427
  Commute             96_820
  Copy             5_907_694
  Erase            7_165_478
  Expand           5_054_907
  Call             5_825_626
  Branch           3_324_179

Memory
  Heap               544_752 B
  Allocated    1_409_188_896 B
  Freed        1_409_188_896 B

[Performance...]
